### PR TITLE
updates

### PR DESCRIPTION
**Pull request by arnetheduck**
* update to latest upstream
* fix exception tracking

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,58 @@
-## description
+## Introduction
 
-**unittest2** is a fork of the [unittest](https://nim-lang.org/docs/unittest.html) module in the
-[Nim](https://nim-lang.org/) standard library, with a focus on [parallel test
-execution](https://status-im.github.io/nim-unittest2/unittest2.html#running-tests-in-parallel).
+**unittest2** started as a [pull request](https://github.com/nim-lang/Nim/pull/9724) to evolve the [unittest](https://nim-lang.org/docs/unittest.html) module in the [Nim](https://nim-lang.org/). It has since grown into a separate library.
```
Comment (stefantalpalaru) [inline README.md:4]: "... in the Nim."

**Comment by stefantalpalaru** [comment]
Looks like the right PR for also setting up CI testing.

**Comment by arnetheduck** [comment]
> Looks like the right PR for also setting up CI testing.

would appreciate some help with that

**Comment by arnetheduck** [comment]
Another thing that would be nice would be to publish the manual to github pages as part of the PR process like https://github.com/status-im/nim-style-guide/blob/main/.github/workflows/gh-pages.yml instead of having the html in the repo.

File: unittest2.nim [hunk 2]
```diff
@@ -218,17 +225,28 @@ var
...
+  nimUnittestOutputLevel {.strdefine.} = $outputLevelDefault
+  nimUnittestColor {.strdefine.} = "auto" ## auto|on|off
+  nimUnittestAbortOnError {.booldefine.} = false
+
+template deprecateEnvVarHere() =
+  # xxx issue a runtime warning to deprecate this envvar.
+  discard
 
-initLock(formattersLock)
-initLock(testFiltersLock)
```
Comment (stefantalpalaru) [inline unittest2.nim:225]: This broke it on Windows.

**Pull request merged**